### PR TITLE
Breadcrumbs

### PR DESCRIPTION
**Pull request by cschweipert**
Adds the link to the parent to breadcrumps if the breadcrumb has a parent and if the current card is the breadcrumb with the parent.

### CODE REVIEW

**Comment by cschweipert** [comment]
This is work in progress. The tests revealed a bug.

File: mod/deckorate_datasets/set/type/dataset.rb [hunk 1]
```diff
@@ -9,14 +10,38 @@
...
+    ]
+
+    insert_parent_link(breadcrumb_array) if should_insert_parent_link?
+
+    breadcrumb_array
+  end
+
+  private
+
+  def should_insert_parent_link?
```
Comment (ethn) [inline mod/deckorate_datasets/set/type/dataset.rb]: more idiomatic just to say `#insert_parent_link?`

File: mod/deckorate_datasets/set/type/dataset.rb [hunk 2]
```diff
@@ -9,14 +10,38 @@
...
+    card.parent != "" && card
```
Comment (ethn) [inline mod/deckorate_datasets/set/type/dataset.rb]: I don't think card will ever be nil.

You could just do `card.parent.present?`

File: mod/deckorate_datasets/set/type/dataset.rb [hunk 3]
```diff
@@ -9,14 +10,38 @@
...
 
-def parent_dataset_card
-  Card[parent_dataset]
+format :html do
+  def breadcrumb_items
+    breadcrumb_array = [
+      link_to("Home", href: "/"),
+      link_to_card(:dataset, "Datasets"),
+      render_name
+    ]
```
Comment (ethn) [inline mod/deckorate_datasets/set/type/dataset.rb]: you could just say `breadcrumb_array = super`, right? Less duplication, more robust if the original method is edited?

File: mod/deckorate_datasets/set/type/dataset.rb [hunk 4]
```diff
@@ -1,6 +1,7 @@
 include_set Abstract::Thumbnail
 include_set Abstract::DeckorateTabbed
 include_set Abstract::Bookmarkable
+include_set Abstract::Breadcrumbs
```
Comment (ethn) [inline mod/deckorate_datasets/set/type/dataset.rb]: `Abstract::DeckorateTabbed` includes `Abstract::Header`, which includes `Abstract::Breadcrumbs`. So this is fine but not really necessary.

**Review by ethn — CHANGES_REQUESTED** [review]
getting there! Just some very minor tweaks suggested

**Review by ethn — APPROVED** [review]
nice

**Pull request merged**